src/Microsoft.ML.AutoML/Experiment/Experiment.cs

```diff
@@ -56,7 +56,7 @@ internal class Experiment<TRunDetail, TMetrics> where TRunDetail : RunDetail
             _experimentSettings = experimentSettings;
             _metricsAgent = metricsAgent;
             _trainerAllowList = trainerAllowList;
-            _modelDirectory = GetModelDirectory(_experimentSettings.CacheDirectory);
+            _modelDirectory = GetModelDirectory(new DirectoryInfo(Path.Combine(_context.TempFilePath,_experimentSettings.CacheDirectoryName)));
```

Comment:
This isn't going to work with the intended behavior from the XML comments on `CacheDirectoryName`:

> If <see langword="null"/>, models will be kept in memory instead of written to disk.

You will need to pass both `_context` and `_experimentSettings.CacheDirectoryName` into this method, and check `CacheDirectoryName` for null, and return `null`.

Reply:
Ah, good catch. Thanks, I'll get that fixed!

Reply:
Ok, fixed.